aggregator/src/aggregator/aggregation_job_driver.rs

```diff
@@ -641,18 +644,31 @@ where
                 .get(RETRY_AFTER)
                 .map(parse_retry_after)
                 .transpose()?;
-            let resp = AggregationJobResp::get_decoded(http_response.body())
-                .map_err(Error::MessageDecode)?;
+
+            let length = http_response
+                .headers()
+                .get(CONTENT_LENGTH)
+                .map(parse_content_length)
+                .transpose()?;
+
+            let resp = if length == Some(0) {
```

Comment:
The semantics here are a little wonky. This treats responses without a content length as finished, which I'm not happy with.